request_test.go

```diff
@@ -38,6 +38,20 @@ func TestDelete(t *testing.T) {
 	})
 }
 
+func TestPatch(t *testing.T) {
+	t.Run("should make a head http request", func(t *testing.T) {
+		spy := &spyRequestExecutor{}
+
+		cli := requestfy.NewClient(
+			requestfy.ConfigRequestExecuter(spy),
+			requestfy.ConfigBaseURL("http://some-cool-domain.local"),
+		)
```

Comment:
I noticed that this snippet is repeated 3 times in this file, I believe it's time to improve this part. What do you think about transforming the 3 test functions into one with different test cases?

I thought of something like:
```diff
diff --git a/request_test.go b/request_test.go
index bc8c8af..eee7933 100644
--- a/request_test.go
+++ b/request_test.go
@@ -10,1 +53,2 @@
-func TestGet(t *testing.T) {
-	t.Run("should concatenate base and specified URL", func(t *testing.T) {
-		spy := &spyRequestExecutor{}
-		client := requestfy.NewClient(
-			requestfy.ConfigRequestExecuter(spy),
-			requestfy.ConfigBaseURL("http://some-cool-domain.local"),
-		)
-
-		res, err := client.Request().Get("cool/path")
-		assertRequestMethod(t, spy, res, err, http.MethodGet)
-
-		if expected, used := "http://some-cool-domain.local/cool/path", spy.lastRequest.URL.String(); used != expected {
-			t.Errorf("expected '%s' URL, used '%s'", expected, used)
-		}
-	})
-}
-
-func TestDelete(t *testing.T) {
-	t.Run("should make a delete http request", func(t *testing.T) {
-		spy := &spyRequestExecutor{}
-
-		cli := requestfy.NewClient(
-			requestfy.ConfigRequestExecuter(spy),
-			requestfy.ConfigBaseURL("http://some-cool-domain.local"),
-		)
-
-		res, err := cli.Request().Delete("bar/foo")
-		assertRequestMethod(t, spy, res, err, http.MethodDelete)
-	})
-}
-
-func TestPatch(t *testing.T) {
-	t.Run("should make a head http request", func(t *testing.T) {
-		spy := &spyRequestExecutor{}
-
-		cli := requestfy.NewClient(
-			requestfy.ConfigRequestExecuter(spy),
-			requestfy.ConfigBaseURL("http://some-cool-domain.local"),
-		)
-
-		res, err := cli.Request().Head("bar/foo")
-		assertRequestMethod(t, spy, res, err, http.MethodHead)
-	})
-}
+func TestRequests(t *testing.T) {
+	testCases := []struct {
+		path           string
+		expectedMethod string
+		method         func(*requestfy.Request) func(string) (*http.Response, error)
+	}{
+		{
+			path:           "cool/path",
+			expectedMethod: http.MethodGet,
+			method: func(r *requestfy.Request) func(string) (*http.Response, error) {
+				return r.Get
+			},
+		},
+		{
+			path:           "bar/foo",
+			expectedMethod: http.MethodDelete,
+			method: func(r *requestfy.Request) func(string) (*http.Response, error) {
+				return r.Delete
+			},
+		},
+		{
+			path:           "bar/foo",
+			expectedMethod: http.MethodHead,
+			method: func(r *requestfy.Request) func(string) (*http.Response, error) {
+				return r.Head
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(fmt.Sprintf("should make %s http request", testCase.expectedMethod), func(t *testing.T) {
+			spy := &spyRequestExecutor{}
+			client := requestfy.NewClient(
+				requestfy.ConfigRequestExecuter(spy),
+				requestfy.ConfigBaseURL(fakeURL),
+			)
+
+			res, err := testCase.method(client.Request())(testCase.path)
+			assertRequestMethod(t, spy, res, err, testCase.expectedMethod)
+		})
+	}
+}
```

Reply:
Sounds good to me. Fixing it.